### DISCUSSION TITLE
ЛР 6 для ИВТ. Формирование mcause

### DISCUSSION THREAD

**Discussion by BROsandr** [MPSU/exAPS#43] [Вопросы и ответы] [answered]
Почему mcause формируется как 27'h4000000? А не, допустим, 27'h0000000.
![image](https://user-images.githubusercontent.com/53278658/204147175-051ef70c-8cea-4e08-a10a-788eaa6c6490.png)

**Comment by gr33nka** [comment, marked as answer]
Тоже дань стандарту, согласно которому старший бит причины `mcause` должен быть единицей, если причиной является прерывание, а не исключение. Например, при считывании неподдерживаемой инструкции (исключение) должен быть код причины в котором старший бит равен 0.

В данном случае речь о аппаратных асинхронных прерываниях, поэтому код причины формируется как `{1'b1, 26'b0, номер_входа_прерывания}`, или, что тоже самое – `{27'h4000000, номер_входа_прерывания}`.

Более подробно об этом можно прочитать в [спецификации на привилегированные инструкции](https://github.com/riscv/riscv-isa-manual/releases/download/Priv-v1.12/riscv-privileged-20211203.pdf) на странице 38
